### PR TITLE
NEW ATTEMPT: Day/Night

### PR DESCRIPTION
**Pull request by Bamhalazam**
## About The Pull Request
(https://github.com/sojourn-13/sojourn-station/pull/4939) 
ORIGINAL PR:

Mostly the same as the old PR, with notable exceptions:
- MAP CHANGES:
      - Added lights along the road, touched nothing else this time.
      - Changed map areas with intent to make it so the light tubes and fixtures work properly.
- FLORA CHANGES:
      - Trees now despawn at nighttime to avoid visual bugs.
      - _If players during testmerge deem it necessary, bushes will be added to the despawn list._
- CODE CHANGES:
      - Jungle turf opacity set to 0 has been reverted
      - Code that fucks with NVGs has been reverted.
      - Adds a new lighting color for use within gatehouse and perimeter - cold blue light. _**colors.dm change**_.
      - Adds new area - **_Colony Road_**, along with it's own APC next to the gatehouse. This is the area responsible for powering the lights along the road.
      - Made it so that invislights disable themselves at night. This is so random cave entrances do not appear lit at night for no reason.
 
<hr>

- POTENTIAL ISSUES:
      - The frills of grass tiles fall out of the tile's borders, meaning they are unlit and create visual bugs. There is nothing I can do about this without refactoring vision code, which would also mean screwing with NVG code. Minor bug as of now.
      - The aforementioned bushes do contribute to this issue, however they too can be despawned at night.

The immediate area around the colony along with anything near the perimeter is 99% visual bug free, at the least :3
Greyson FO and other dungeons of the type have dyn-lighting disabled to save on performance.

Space tiles aren't an issue anymore as I started fresh.

<hr>

The current state of the road lighting:

_Do note that some areas were **intended** to be unlit to facilitate sneaking, and that the view range in this pic is 128x128, it looks better with normal range!_

itiswhatitis

<hr>
</details>

## Changelog

add: Added more things
fix: fixed a few things
code: changed some code
refactor: refactored some code

### CODE REVIEW

**Comment by Bamhalazam** [comment]
This fucking code just _isn't built for this. man!!!_

**Comment by Trilbyspaceclone** [comment]
Can you get a screen shot of it being a day shift to make sure its all working under "normal" cases

**Comment by Bamhalazam** [comment]
> Can you get a screen shot of it being a day shift to make sure its all working under "normal" cases
Here it is @Trilbyspaceclone 
image

- **ADDENDUM:**
        - Since I forgot to change typepaths to include areas like the gatehouse and perimeter in the dynamic lighting proc earlier, I did them now. Areas such as _/area/nadezhda/security/maingate_ are now _/area/nadezhda/**outside**/security/maingate_ to facilitate that. 
        - Additionally, invislights have been added to structures such as the burnt outpost, and the farm to give the illusion of light falling into them from the outside, due to them now being dynamically lit too.
        
image
image

**Comment by Bamhalazam** [comment]
Note to Self:

- Crater Deep Jungle entrance needs invislight
- Render pit needs dynlight

**Comment by Bamhalazam** [comment]
- Scrap haven needs dynlight

**Comment by FoxBopGaming** [comment]
image
this area in the gatehouse could use a light

**Comment by benj8560** [comment]
Noticing a lotta issues with moving to areas with my entire screen turning white. In some cases not exactly consistent but jumping into AI shells/AI drones seems to do it a lot. 

The bushes are **horrendous** and **need** a fix. Foke with small scaled sprites can currently hide under them effortlessly and it makes it a nightmare to track them.

**Comment by Bamhalazam** [comment]
> Noticing a lotta issues with moving to areas with my entire screen turning white. In some cases not exactly consistent but jumping into AI shells/AI drones seems to do it a lot
.
That's unfortunately lag related, due to the dynlighting being on. I'll see about disabling it in more spots to save on performance. 

> The bushes are **horrendous** and **need** a fix. Foke with small scaled sprites can currently hide under them effortlessly and it makes it a nightmare to track them.
Are the small bushes layered over them? I forgot from yesterday XD

**Comment by benj8560** [comment]
>That's unfortunately lag related what with the box being unable to handle it, since the dyn-lighting is on. I'll see about disabling it in more spots to save on performance.

Thing is it sometimes sticks around.....some foke gotta go as far as reconnecting or moving to a new area to get it to clear it.

> Are the small bushes layered over them? I forgot from yesterday XD

Just spawn in as anything on a private server or on live. Run around just inside the fence you'll see exactly what i mean

**Comment by benj8560** [comment]
(kinda hard to move to a new area when your nearly blind)

**Comment by benj8560** [comment]
it's /obj/structure/flora/big/rocks3 now that I've gotten outta bed finally

**Comment by Bamhalazam** [comment]
> so afaik, the trees were removed because of a brightness issue? I bring this up because I found a wild, lone tree, yet it seems completely fine and natural, is this something unique to this type of tree? or is this a bug or something which was overlooked?? Are Yggdrasil's immune to the lighting fuckery? Or was the lighting fuckery a figment of imagination? image


Yggdrasil is under a separate typepath, thus it escapes the wrath of the tree removal and it is rarely enough seen to not be an issue! I will look into making it emit light

**Comment by Bamhalazam** [comment]
@cdb-is-not-good pls merge

**Comment by Bamhalazam** [comment]
For the record I am aware of the scrap haven lower floor issue but I am unable to fix it before it gets fullmerged due to repo fuckery

**Comment by Bamhalazam** [comment]
> image this area in the gatehouse could use a light

added!

File: code/__HELPERS/time.dm [hunk 1]
```diff
@@ -205,3 +205,10 @@ var/global/rollovercheck_last_timeofday = 0
 	while (world.tick_usage > min(TICK_LIMIT_TO_RUN, Master.current_ticklimit))
 
 #undef DELTA_CALC
+
+proc/determine_night() // Determines whether it's night or not and returns TRUE or FALSE accordingly
+	if(true_roundstart_hour <= 20 || true_roundstart_hour >= 8)
```
Comment (SingingSpock) [inline code/__HELPERS/time.dm:210]:
```suggestion
	if(true_roundstart_hour <= 20 && true_roundstart_hour >= 8)
```

**Comment by FoxBopGaming** [comment]
> Theyre super fucked up bruh
> 
> exampletree

That's how they looked? honestly that doesn't feel nearly as horrid as it was made out to sound, thats as far as i'm concerned, pretty alright, you made it sound like a fullbright hell, not a dimmed sprite overlapping pure black

**Comment by benj8560** [comment]
Currently the performance hit is just to great for me to like this. I love everything else about it but...I'm stuttering around constantly at night the lag is atrocious and I gotta assume this is why.

**Comment by AysheDaArtist** [comment]
Is it possible to change the trees to behave like bushes, and just add in code that when a saw is hit on a 'tree' it cuts down wood.

Love the nighttime stuff, I haven't really noticed a performance hit, but I'd say it does impact the server with all the lighting ( from a technical stand point ) but personally I think this adds a lot to Sojourn's events and  setting

**Comment by Bamhalazam** [comment]
> Is it possible to change the trees to behave like bushes, and just add in code that when a saw is hit on a 'tree' it cuts down wood.

this is already a thing tho..?
:3

**Comment by Trilbyspaceclone** [comment]
Closing do to conflict merge issues.
If you wish to reopen a new one with corrected map/coding please do so.
This pr **is not denied** at the type of making this commit, it is being cleared do to staleness

**Pull request closed without merging**